### PR TITLE
Fix layers losing their names upon being grouped

### LINKED ISSUE #1644: Layers lose their names when grouped

### PR DESCRIPTION
**Pull request by moOsama76**
Make layers reserve their visible names when grouped together (CTRL+G)

Closes #1644

### CODE REVIEW

File: editor/src/messages/portfolio/document/document_message_handler.rs [hunk 1]
```diff
@@ -419,7 +420,13 @@ impl MessageHandler<DocumentMessage, DocumentInputs<'_>> for DocumentMessageHand
 
 					let id = NodeId(generate_uuid());
 					let insert_index = -1;
-					responses.add(GraphOperationMessage::NewCustomLayer { id, nodes, parent, insert_index });
+					responses.add(GraphOperationMessage::NewCustomLayer {
+						id,
+						nodes,
+						parent,
+						insert_index,
+						alias: String::from(""),
```
Comment (TrueDoctor) [inline editor/src/messages/portfolio/document/document_message_handler.rs:428]: You can use `String::new()` instead, which does not allocate memory

Comment (moOsama76) [reply editor/src/messages/portfolio/document/document_message_handler.rs:428]: Sure

Comment (moOsama76) [reply editor/src/messages/portfolio/document/document_message_handler.rs]: done

Comment (0HyperCube) [reply editor/src/messages/portfolio/document/document_message_handler.rs]: @TrueDoctor I'm pretty sure `String::from("")` won't allocate - it looks like it just transmutates the string into a byte slice and then converts it to a Vec which will not allocate if the size is 0.

**Review by 0HyperCube — APPROVED** [review]
Works well - thanks for this fix.

**Comment by Keavon** [comment]
@moOsama76 please resolve the merge conflict (and in the future, it'll be easier if you make sure you are always rebasing your ongoing work with master on a frequent basis to reduce the conflicts that arise at the end). Thanks!

**Comment by Keavon** [comment]
Actually I went ahead and did it for you since it was faster.

File: editor/src/messages/portfolio/document/node_graph/graph_operation_message_handler.rs [hunk 2]
```diff
@@ -689,6 +695,8 @@ impl MessageHandler<GraphOperationMessage, GraphOperationHandlerData<'_>> for Gr
 						})
 						.unwrap_or_default();
 
+					modify_inputs.document_network.nodes.get_mut(&id).unwrap().alias = alias.clone();
```
Comment (Keavon) [inline editor/src/messages/portfolio/document/node_graph/graph_operation_message_handler.rs:698]: Please avoid this unwrap, perhaps using `if let`.

File: editor/src/messages/portfolio/portfolio_message_handler.rs [hunk 3]
```diff
@@ -180,10 +180,10 @@ impl MessageHandler<PortfolioMessage, (&InputPreprocessorMessageHandler, &Prefer
 						};
 
 						let node = layer.to_node();
+						let previous_alias = active_document.network().nodes.get(&node).unwrap().alias.clone();
```
Comment (Keavon) [inline editor/src/messages/portfolio/portfolio_message_handler.rs:183]: Please avoid this unwrap, perhaps defaulting to an empty string.

File: editor/src/messages/portfolio/document/node_graph/graph_operation_message_handler.rs (hunk 2 above)
Comment (Keavon) [reply editor/src/messages/portfolio/document/node_graph/graph_operation_message_handler.rs]: I made this change, but please look at the code to learn from the updated code for your future reference.

File: editor/src/messages/portfolio/portfolio_message_handler.rs (hunk 3 above)
Comment (Keavon) [reply editor/src/messages/portfolio/portfolio_message_handler.rs]: I made this change, but please look at the code to learn from the updated code for your future reference.

**Comment by Keavon** [comment]
!build

**Comment by moOsama76** [comment]
Thank you, I was going to fix but my partition get full and I had some cache problems

File: editor/src/messages/portfolio/document/document_message_handler.rs (hunk 1 above)
Comment (TrueDoctor) [reply editor/src/messages/portfolio/document/document_message_handler.rs]: @0HyperCube  In debug mode it allocates, when compiling with optimizations the same assembly is produced so I guess it is just good practice to use `String::new()` to guarantee it won't allocate + speed up debug builds but will most likely not make difference in production: https://godbolt.org/z/9r5o7jrs4

**Pull request merged**